cli/attestation.go

```diff
@@ -473,17 +554,183 @@ func (cli *CLI) NewValidateAttestationValidationCmd() *cobra.Command {
+	tpmAttestationFile = string(args[0])
+	attestationBytes, err := io.ReadAll(dataInput())
+	if err != nil {
+		return err
+	}
+	attestation := &tpmAttest.Attestation{}
+
+	if format == "binarypb" {
+		err = proto.Unmarshal(attestationBytes, attestation)
+	} else if format == "textproto" {
+		err = unmarshalOptions.Unmarshal(attestationBytes, attestation)
+	} else {
+		return fmt.Errorf("format should be either binarypb or textproto")
+	}
+	if err != nil {
+		return fmt.Errorf("fail to unmarshal attestation report: %v", err)
+	}
+
+	pub, err := tpm2.DecodePublic(attestation.GetAkPub())
+	if err != nil {
+		return err
+	}
+	cryptoPub, err := pub.Key()
+	if err != nil {
+		return err
+	}
+
+	var validateOpts interface{}
+	switch attestation.GetTeeAttestation().(type) {
+	case *tpmAttest.Attestation_SevSnpAttestation:
+		if len(teeNonce) != 0 {
+			validateOpts = &server.VerifySnpOpts{
+				Validation:   server.SevSnpDefaultValidateOpts(teeNonce),
+				Verification: &sevVerify.Options{},
+			}
+		} else {
+			validateOpts = &server.VerifySnpOpts{
+				Validation:   server.SevSnpDefaultValidateOpts(nonce),
+				Verification: &sevVerify.Options{},
+			}
+		}
+	default:
+		validateOpts = nil
+	}
+
+	ms, err := server.VerifyAttestation(attestation, server.VerifyOpts{Nonce: nonce, TrustedAKs: []crypto.PublicKey{cryptoPub}, TEEOpts: validateOpts})
+	if err != nil {
+		return fmt.Errorf("verifying attestation: %w", err)
+	}
+	out, err := marshalOptions.Marshal(ms)
+	if err != nil {
 		return nil
 	}
+	if _, err := dataOutput().Write(out); err != nil {
+		return fmt.Errorf("failed to write verified attestation report: %v", err)
+	}
 
-	if err := cmd.MarkFlagRequired("product"); err != nil {
-		printError(cmd, "Failed to mark flag as required: %v ❌ ", err)
+	return nil
+}
+
+func vtpmverify(args []string) error {
+	tpmAttestationFile = string(args[0])
+	attestationBytes, err := io.ReadAll(dataInput())
+	if err != nil {
+		return err
+	}
+	attestation := &tpmAttest.Attestation{}
+
+	if format == "binarypb" {
+		err = proto.Unmarshal(attestationBytes, attestation)
+	} else if format == "textproto" {
+		err = unmarshalOptions.Unmarshal(attestationBytes, attestation)
+	} else {
+		return fmt.Errorf("format should be either binarypb or textproto")
+	}
+	if err != nil {
+		return fmt.Errorf("fail to unmarshal attestation report: %v", err)
+	}
+
+	pub, err := tpm2.DecodePublic(attestation.GetAkPub())
+	if err != nil {
+		return err
+	}
+	cryptoPub, err := pub.Key()
+	if err != nil {
+		return err
+	}
+
+	ms, err := server.VerifyAttestation(attestation, server.VerifyOpts{Nonce: nonce, TrustedAKs: []crypto.PublicKey{cryptoPub}, TEEOpts: nil})
+	if err != nil {
+		return fmt.Errorf("verifying attestation: %w", err)
+	}
+	out, err := marshalOptions.Marshal(ms)
+	if err != nil {
 		return nil
 	}
+	if _, err := dataOutput().Write(out); err != nil {
+		return fmt.Errorf("failed to write verified attestation report: %v", err)
+	}
```

Comment:
```suggestion
    output, err := createOutputFile()
    if err != nil {
        return err
    }
    if closer, ok := output.(*os.File); ok {
        defer closer.Close()
    }
	if _, err := output.Write(out); err != nil {
		return fmt.Errorf("failed to write verified attestation report: %v", err)
	}
```

Reply:
this will remove need for the alwaysError struct which does not seem idiomatic in go